python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -442,44 +430,36 @@ def write_parquet(
     --------
     cudf.io.parquet.write_parquet
     """
-
-    # Create the write options
-    cdef table_input_metadata tbl_meta
-
-    cdef vector[map[string, string]] user_data
-    cdef table_view tv
-    cdef vector[unique_ptr[data_sink]] _data_sinks
-    cdef sink_info sink = make_sinks_info(
-        filepaths_or_buffers, _data_sinks
-    )
-
     if index is True or (
         index is None and not isinstance(table._index, cudf.RangeIndex)
     ):
-        tv = table_view_from_table(table)
-        tbl_meta = table_input_metadata(tv)
+        columns = table._columns if table._index is None else [
```

Comment:
`table._index` should (ideally) never be `None`. Is there a test failure if we don't check for this?

Reply:
Thanks, no test failures. I removed the checks.